tools/lib/url_file.py

```diff
@@ -35,13 +36,13 @@ def __init__(self, url, debug=False, cache=None):
     if cache is not None:
       self._force_download = not cache
 
-    try:
-      self._curl = self._tlocal.curl
-    except AttributeError:
-      self._curl = self._tlocal.curl = pycurl.Curl()
     if not self._force_download:
       os.makedirs(Paths.download_cache_root(), exist_ok=True)
 
+    if not hasattr(URLFile._tlocal, 'http_client'):
```

Comment:
the try/except is more thread safe

Reply:
like this?
```
try:
  self._http_client = URLFile._tlocal.http_client
except AttributeError:
  URLFile._tlocal.http_client = PoolManager()
  self._http_client = URLFile._tlocal.http_client
```

Reply:
why not just copy the way the curl one was done?